src/python/ensembl/io/genomio/assembly/status.py

```diff
@@ -0,0 +1,478 @@
+) -> Union[Tuple[Dict, str]]:
+    """Function to identify the kind of queries being passed by user,
+    then extract the queries (core names or accessions) and store each with appropriate identifier.
+
+    Args:
+        query_list: List of user defined queries either core names, or accessions
+        partial_url: A partial MYSQL connection URL (host:port)
+        input_cores: arg parse param '--input_cores'
+        input_accessions: arg parse param '--input_accns'
+
+    Returns:
+        User queries stored as identifier[(core db name | UniqueID#)] : accession
+    """
+
+    query_accessions: Dict = {}
+    query_type: str = ""
+
+    if input_cores and input_accessions is None:
+        query_accessions = fetch_accessions_from_cores(query_list, partial_url)
+        query_type = "CoreDB"
+    elif input_cores is None and input_accessions:
+        query_type = "Accession"
+        for accession in query_list:
+            match = re.match(r"(GC[AF])_([0-9]{3})([0-9]{3})([0-9]{3})\.?([0-9]+)", accession)
+            if not match:
+                raise UnsupportedFormatError(f"Could not recognize GCA accession format: {accession}")
+            query_accessions[accession] = accession
+
+    return query_accessions, query_type
+
+
+def fetch_accessions_from_cores(database_names: list, connection_url: URL) -> Dict:
+    """Obtain the associated INSDC accession [meta.assembly.accession] given a set of core(s) names
+    and a MYSQL server host.
+
+    Args:
+        database_names: Set of names for one or more core databases
+        connection_url: Partial MYSQL host name : port
+    Returns:
+        Dict of core name(s) (key) and its INSDC assembly.accession (value)
+    """
+
+    core_accn_meta = {}
+    core_list_count = len(database_names)
+    count_accn_found = 0
+
+    for core in database_names:
+        db_connection_url = connection_url.set(database=core)
+        db_connection = dbc(f"{db_connection_url}")
+        qry_result = db_connection.execute(
+            'SELECT meta_value FROM meta WHERE meta_key = "assembly.accession";'
+        ).fetchall()
+
+        if qry_result is None:
+            logging.warning(f"We have no accession on core: {core}")
+        elif len(qry_result) == 1:
+            count_accn_found += 1
+            asm_accession = qry_result.pop()[0]
+            logging.info(f"{core} -> assembly.accession[{asm_accession}]")
+            core_accn_meta[core] = asm_accession
+        else:
+            logging.warning(f"Core {core} Has {len(qry_result)} assembly.accessions")
+
+    logging.info(f"From initial input cores ({core_list_count}), obtained ({count_accn_found}) accessions")
+
+    return core_accn_meta
+
+
+def datasets_asm_reports(
+    sif_image: str, assembly_accessions: dict, download_directory: PathLike, batch_size: int
+) -> Dict:
+    """Obtain multiple assembly report JSONs in one or more querys to datasets,
+    i.e. make individual since accn query to datasets tool.
```

Comment:
I'm afraid I cannot understand this sentence as it is written.

Reply:
Indeed word jumble. Fix inbound